### PR TITLE
add additional test data set for nuget pipeline

### PR DESCRIPTION
**Pull request by askhade**
**Description**: 
1. Adding additional test data set for Nuget packaging pipelines. 
2. Update download test data script to pick build secrets and use this secret in the download request

### CODE REVIEW

**Comment by yuslepukhin** [comment]
Please, describe your changes.

**Comment by askhade** [comment]
> Please, describe your changes.

Marked it as WIP... I will add description once this is ready to be reviewed. Changes work on my machine but ironing out some issues in the pipelines

File: tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ jobs:
   pool: 
     vmImage: 'macOS-10.13'
   steps:
-  
+    - template: templates/set-test-data-variables-step.yml
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml]: Do you need to also update the python packaging pipeline ?

Comment (askhade) [reply tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml]: No I dont... actually now I can undo this change too...

Comment (snnn) [reply tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml]: It's ok.

**Comment by askhade** [comment]
Note: I queued CPI packaging pipeline for this PR because in one my commits I had made a small change to this pipeline, however I ended up undoing that change. Current CPI packaging failure in not related to the change on this PR hence no blocking the merge on it.

**Pull request merged**